ui/component/claimList/view.jsx

```diff
@@ -243,6 +249,7 @@ export default function ClaimList(props: Props) {
           tileUris.map((uri, index) => {
             if (uri) {
               const inj = getInjectedItem(index);
+              if (inj) refreshBuffer();
```

Comment:
Calling it here in the render function means we are setting state while rendering.  While it is permitted, there is a chance of ending up in infinite loop.

Taking a step back to find a better solution.  Some questions:

- Between `if (inj) refreshBuffer();` and next `return` statement where `uriBuffer` is used ... in the `return` statement, do you expect `uriBuffer` to be the latest data updated in the `if` statement, or using data from the previous cycle?  (The answer can actually only be the latter, since set-state is not immediate and instant update is not guaranteed.  So, I'll assume that).
- But if it's needs data from the previous cycle, then it should only be done in an effect, so seems like you need to use the new data in the next index of the loop, while maintaining the end result for the next cycle?
- In that case, that a look at `useRef`
  - Notes for refs:  
    - refs does not spark a render when changed.
    - Remember that it needs to access the `.current` property.

Reply:
Jep, this `refreshBuffer` function is the reason why I wanted you to take a look at this. I remember you telling me that this should be avoided if possible last time. However, I wasn't able to fix this in an effect because the actual change isn't available in any variable that would be accessible from outside of the loop. The `inj` object only exists inside of the loop and does not effect any other variable.

...unless I'd somehow trigger a change in the `getInjectedItem` function, but that wouldn't solve this issue and still call an update from within the loop.

This current solution works and I don't think it can trigger an infinite loop unless I'm missing something. Updating the buffer does not result in a redraw. It can only reduce the number of cycles in the loop. Please let me know if you know if there's a better way to do this.

Reply:
Ah, I see now why it "worked" and it didn't cause render issues -- another violation masked the problem:  ["don't mutate states directly"](https://dev.to/il3ven/common-error-accidentally-mutating-state-in-react-4ndg#:~:text=In%20React%2C%20the%20state%20is,set%20the%20state%20using%20setState%20.) (i.e. `.push` mutated the original buffer since it was assigned instead of copied/sliced).

Essentially, you are using it like a ref, so I think we can just change it from `useState` to `useRef`, remove the need for the effect (if I understand the usage), and make necessary adjustments (like doing `.current` to access the ref's value).

Reply:
When I use `useRef` for the `uriBuffer` instead it does trigger an infinite loop...

Reply:
See if this covers everything, or if I broke any logic:
https://github.com/lbryio/lbry-desktop/commit/ce8bb801f20e3afe6f79167df5a521a15411f1b1